### PR TITLE
Feature/151934 build contacts pages

### PR DESCRIPTION
**Pull request by danielryannimble**
contacts pages

api controller and services all added

### CODE REVIEW

File: Dfe.ManageFreeSchoolProjects/Dfe.ManageFreeSchoolProjects/Pages/Project/Contacts/EditSchoolChairContactPage.cshtml [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+<div class="govuk-grid-row">
+    <div class="govuk-grid-column-full">
+        <span class="govuk-caption-l" data-testid="school-name">@Model.SchoolName</span>
+        <h1 class="govuk-heading-xl">
+            @ViewData["Title"]
+        </h1>
+    </div>
+</div>
+
+<div class="govuk-grid-row">
```
Comment (sukhybhullar-nimble) [inline Dfe.ManageFreeSchoolProjects/Dfe.ManageFreeSchoolProjects/Pages/Project/Contacts/EditSchoolChairContactPage.cshtml:30]: Is this empty div supposed to be here?

File: Dfe.ManageFreeSchoolProjects/Dfe.ManageFreeSchoolProjects/Pages/Project/Contacts/EditSchoolChairContactPage.cshtml.cs [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+    private readonly IGetProjectOverviewService _getProjectOverviewService;
+    
+    private readonly ILogger<EditSchoolChairContactModel> _logger;
+    
+    private readonly ErrorService _errorService;
+    
+    [BindProperty(SupportsGet = true, Name = "projectId")]
+    public string ProjectId { get; set; }
+    
+    [BindProperty(SupportsGet = true, Name = "school-chair-name")]
```
Comment (sukhybhullar-nimble) [inline Dfe.ManageFreeSchoolProjects/Dfe.ManageFreeSchoolProjects/Pages/Project/Contacts/EditSchoolChairContactPage.cshtml.cs:34]: I dont this we need supports get for this - unless I'm missing something?

File: Dfe.ManageFreeSchoolProjects/Dfe.ManageFreeSchoolProjects/Pages/Project/Contacts/EditSchoolChairContactPage.cshtml.cs [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+    [ValidText(100)]
+    [DisplayName("School chair name")]
+    public string SchoolChairName { get; set; }
+
+    [BindProperty(SupportsGet = true, Name = "school-chair-email")]
```
Comment (sukhybhullar-nimble) [inline Dfe.ManageFreeSchoolProjects/Dfe.ManageFreeSchoolProjects/Pages/Project/Contacts/EditSchoolChairContactPage.cshtml.cs:39]: I dont this we need supports get for this - unless I'm missing something?

File: Dfe.ManageFreeSchoolProjects/Dfe.ManageFreeSchoolProjects/Pages/Project/Contacts/EditSchoolChairContactPage.cshtml.cs [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+        };
+
+        await _addContactsService.Execute(ProjectId, updateContactsRequest);
+
+        return Redirect(GetNextPage());
+    }
+    
+    private static bool IsEmailValid(string email)
+    {
+        return email != null && new EmailAddressAttribute().IsValid(email);
```
Comment (sukhybhullar-nimble) [inline Dfe.ManageFreeSchoolProjects/Dfe.ManageFreeSchoolProjects/Pages/Project/Contacts/EditSchoolChairContactPage.cshtml.cs:149]: Does this cause the email to be required: i.e. if email is blank will it cause an error.

File: Dfe.ManageFreeSchoolProjects/Dfe.ManageFreeSchoolProjects/Pages/Project/Contacts/EditTrustChairContactPage.cshtml [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+<div class="govuk-grid-row">
+    <div class="govuk-grid-column-full">
+        <h1 class="govuk-heading-xl">
+              <span class="govuk-caption-l" data-testid="school-name">@Model.SchoolName</span>
+            @ViewData["Title"]
+        </h1>
+    </div>
+</div>
+
+<div class="govuk-grid-row">
```
Comment (sukhybhullar-nimble) [inline Dfe.ManageFreeSchoolProjects/Dfe.ManageFreeSchoolProjects/Pages/Project/Contacts/EditTrustChairContactPage.cshtml:30]: Dont think this empty div is supposed to be here

File: Dfe.ManageFreeSchoolProjects/Dfe.ManageFreeSchoolProjects/Pages/Project/Contacts/EditTrustChairContactPage.cshtml.cs [hunk 6]
```diff
@@ -0,0 +1,151 @@
...
+    private readonly IGetProjectOverviewService _getProjectOverviewService;
+    
+    private readonly ILogger<EditTrustChairContactModel> _logger;
+    
+    private readonly ErrorService _errorService;
+    
+    [BindProperty(SupportsGet = true, Name = "projectId")]
+    public string ProjectId { get; set; }
+    
+    [BindProperty(SupportsGet = true, Name = "trust-chair-name")]
```
Comment (sukhybhullar-nimble) [inline Dfe.ManageFreeSchoolProjects/Dfe.ManageFreeSchoolProjects/Pages/Project/Contacts/EditTrustChairContactPage.cshtml.cs:34]: This shouldn't need supports get

File: Dfe.ManageFreeSchoolProjects/Dfe.ManageFreeSchoolProjects/Pages/Project/Contacts/EditTrustChairContactPage.cshtml.cs [hunk 7]
```diff
@@ -0,0 +1,151 @@
...
+    [ValidText(100)]
+    [DisplayName("Trust chair name")]
+    public string TrustChairName { get; set; }
+
+    [BindProperty(SupportsGet = true, Name = "trust-chair-email")]
```
Comment (sukhybhullar-nimble) [inline Dfe.ManageFreeSchoolProjects/Dfe.ManageFreeSchoolProjects/Pages/Project/Contacts/EditTrustChairContactPage.cshtml.cs:39]: This shouldn't need supports get

**Review by sukhybhullar-nimble — CHANGES_REQUESTED** [review]

**Review by sukhybhullar-nimble — APPROVED** [review]

File: Dfe.ManageFreeSchoolProjects/Dfe.ManageFreeSchoolProjects/Pages/Project/Contacts/EditTrustChairContactPage.cshtml.cs (hunk 7 above)
Comment (danielryannimble) [reply Dfe.ManageFreeSchoolProjects/Dfe.ManageFreeSchoolProjects/Pages/Project/Contacts/EditTrustChairContactPage.cshtml.cs]: resolved

File: Dfe.ManageFreeSchoolProjects/Dfe.ManageFreeSchoolProjects/Pages/Project/Contacts/EditTrustChairContactPage.cshtml.cs (hunk 6 above)
Comment (danielryannimble) [reply Dfe.ManageFreeSchoolProjects/Dfe.ManageFreeSchoolProjects/Pages/Project/Contacts/EditTrustChairContactPage.cshtml.cs]: sorted

File: Dfe.ManageFreeSchoolProjects/Dfe.ManageFreeSchoolProjects/Pages/Project/Contacts/EditTrustChairContactPage.cshtml (hunk 5 above)
Comment (danielryannimble) [reply Dfe.ManageFreeSchoolProjects/Dfe.ManageFreeSchoolProjects/Pages/Project/Contacts/EditTrustChairContactPage.cshtml:30]: sorted

File: Dfe.ManageFreeSchoolProjects/Dfe.ManageFreeSchoolProjects/Pages/Project/Contacts/EditSchoolChairContactPage.cshtml.cs (hunk 2 above)
Comment (danielryannimble) [reply Dfe.ManageFreeSchoolProjects/Dfe.ManageFreeSchoolProjects/Pages/Project/Contacts/EditSchoolChairContactPage.cshtml.cs]: resolved

File: Dfe.ManageFreeSchoolProjects/Dfe.ManageFreeSchoolProjects/Pages/Project/Contacts/EditSchoolChairContactPage.cshtml (hunk 1 above)
Comment (danielryannimble) [reply Dfe.ManageFreeSchoolProjects/Dfe.ManageFreeSchoolProjects/Pages/Project/Contacts/EditSchoolChairContactPage.cshtml:30]: resolved

**Pull request merged**